servers/physics_2d/step_2d_sw.cpp

```diff
@@ -31,19 +31,23 @@
 #include "step_2d_sw.h"
 #include "core/os/os.h"
 
-void Step2DSW::_populate_island(Body2DSW *p_body, Body2DSW **p_island, Constraint2DSW **p_constraint_island) {
+#define BODY_ISLAND_COUNT_RESERVE 128
+#define BODY_ISLAND_SIZE_RESERVE 512
+#define ISLAND_COUNT_RESERVE 128
+#define ISLAND_SIZE_RESERVE 512
+
+void Step2DSW::_populate_island(Body2DSW *p_body, LocalVector<Body2DSW *> &p_body_island, LocalVector<Constraint2DSW *> &p_constraint_island) {
 	p_body->set_island_step(_step);
-	p_body->set_island_next(*p_island);
-	*p_island = p_body;
+	p_body_island.push_back(p_body);
 
-	for (const List<Pair<Constraint2DSW *, int>>::Element *E = p_body->get_constraint_list().front(); E; E = E->next()) {
+	// Faster with reversed iterations.
```

Comment:
Interesting, if you can add why it's faster, it would be useful.

Reply:
The truth is I'm not sure exactly why, but I could see a clear difference in all the tests I've run :D
I'll check a bit more if I can figure it out, otherwise I'll leave a note with a TODO to investigate that more later.

Reply:
hmm that's even more interesting then. Most likely this improvement is just related to this specific project, so if you change project setup the forward may be faster. I think that this highlight a possible optimization: If you can understand what's the determining factor, you could consider to add a sorting mechanism (somehow) so you can apply this optimization to any project.

Reply:
Yeah that makes sense. I agree it would be interesting to investigate that specifically with different use cases. For now I'm keeping a reversed order, because it's closer to what happened before this PR, due to how islands where fed in a reversed order (each new pair was set as the new root, and the previous root was set as the next pair).